src/sql/src/plan/statement/ddl.rs

```diff
@@ -424,6 +432,13 @@ pub fn plan_create_webhook_source(
         in_cluster,
     } = stmt;
 
+    for ident in name.0.iter() {
+        let name_str = ident.as_str();
+        if !is_name_valid(name_str) {
+            return Err(PlanError::InvalidWebhookName);
+        }
+    }
+
```

Comment:
This will prohibit webhooks with invalid names, but not webhooks in databases or schemas with invalid names.

It also occurs to me that we need to protect against:

  * Renaming a database containing webhook sources to `.` or `..`
  * Renaming a schema containing webhook sources to `.` or `..`
  * Renaming a webhook source itself to `.` or `..`

Reply:
That's going to be very complicated, though. Perhaps it really would be better to simply ban `.` and `..` across the board. That's more aggressive than necessary, but it means we don't need to add special cases for renaming.

Reply:
I like it. This would move to CreateSourcePlan (~line 563 in ddl.rs)?

Reply:
To apply the restriction universally, I think we actually want to make it a check in `Ident::new`: https://github.com/MaterializeInc/materialize/blob/d6ca1e1276bd0153075b5c9d5738bd75436a1be1/src/sql-parser/src/ast/defs/name.rs#L36-L64